src/libraries/Directory.Build.props

```diff
@@ -256,4 +256,7 @@
     <CLSCompliant Condition="'$(CLSCompliant)' == ''">true</CLSCompliant>
   </PropertyGroup>
 
+  <PropertyGroup Condition="$([MSBuild]::IsOSPlatform('windows')) and '$(TargetOS)' == 'Browser'"> <!-- the Unix versions are used without this override -->
```

Comment:
Is the actual problem a bug in `eng\Configurations.props`?

I think the block in Configurations.props should be something like:
```
    <!-- There are no WebAssembly tools, so use the default ones -->
    <_toolRuntimeRID Condition="'$(_runtimeOS)' == 'browser'">linux-x64</_toolRuntimeRID>
    <_toolRuntimeRID Condition="'$(_runtimeOS)' == 'browser' and $([MSBuild]::IsOSPlatform('WINDOWS'))">win-x64</_toolRuntimeRID>
    <_toolRuntimeRID Condition="'$(_runtimeOS)' == 'browser' and $([MSBuild]::IsOSPlatform('OSX'))">osx-x64</_toolRuntimeRID>
```

Reply:
Thanks, that looks better.  Admittedly I don't know my way around the build files.

Reply:
This commit was made before the configurations were even moved to eng/Configurations.props. And it is/was not a bug.

The new thing this PR has introduced is `TargetOS=browser` set from cmd file, we typically do not add TargetOS in cmd/sh for cross-compilation scenarios, where it has specific meaning. If we change cmd/sh files here with new `TargetOS=browser` introducing new meaning, sooner or later that will clash with mono, where we are using common paths in scripts. It is anyway confusing.

If we do not introduce TargetOS=browser inside the script and instead take mono's lead on making 'is browser' decision, you would be able to take benefit of many common work without configuration changes over time in a smooth way.

Reply:
This condition should be using the OS that the build is running on.

TargetOS should mean the OS we are compiling for, everywhere in the build system. If TargetOS means the OS that the build is running in some places, I think we should fix those place.

> mono's lead on making 'is browser' decision

I am not sure what this means.

Reply:
TargetOS is not `browser` on mono when we are building for wasm today (it has been this way all along). If we intend to change that as a fix, OK.

Reply:
> TargetOS is not browser on mono when we are building for wasm today

Is it really the case? There are number of places under libraries that compare TargetOS with Browser, one example from many: https://github.com/dotnet/runtime/blob/master/src/libraries/System.Diagnostics.StackTrace/tests/System.Diagnostics.StackTrace.Tests.csproj#L16 

If the TargetOS was not set correct for Browser, all these places would be broken.

In any case, I think it is a good idea to keep fixing places that use wrong meaning of TargetOS.